src/main/java/com/example/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package com.example.service.impl;
+
+import com.example.dto.cart.CreateCartItemRequestDto;
+import com.example.dto.cart.ShoppingCartDto;
+import com.example.dto.cart.UpdateCartItemRequestDto;
+import com.example.exception.EntityNotFoundException;
+import com.example.exception.ProcessingException;
+import com.example.mapper.CartItemMapper;
+import com.example.mapper.ShoppingCartMapper;
+import com.example.model.Book;
+import com.example.model.CartItem;
+import com.example.model.ShoppingCart;
+import com.example.repository.book.BookRepository;
+import com.example.repository.cart.CartItemRepository;
+import com.example.repository.cart.ShoppingCartRepository;
+import com.example.service.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCartDto getCartByUserId(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId).orElseThrow(
+                () -> new EntityNotFoundException(
+                        "Can't find shopping cart by user id = " + userId));
+        ShoppingCartDto shoppingCartDto = shoppingCartMapper.toDto(shoppingCart);
+        shoppingCartDto.setCartItems(shoppingCart.getCartItems().stream()
+                .map(cartItemMapper::toDto)
+                .collect(Collectors.toSet()));
+        return shoppingCartDto;
```

Comment:
```suggestion
        return shoppingCartMapper.toDto(shoppingCartRepository.findByUserId(userId));
```

where in ShoppingCartRepository

```
@EntityGraph(attributePaths = {"cartItems", "cartItems.book"})
Optional<ShoppingCart> findByUserId(Long userId);
```